app/src/test/java/org/simple/clinic/scanid/ScanSimpleIdUpdateTest.kt

```diff
@@ -36,6 +51,50 @@ class ScanSimpleIdUpdateTest {
         ))
   }
 
+  @Test
+  fun `when the qr code is scanned and the returned string is json then parse the json into object`() {
+    spec
+        .given(defaultModel)
+        .whenEvent(ScanSimpleIdScreenQrCodeScanned(expectedJson))
+        .then(assertThatNext(
+            hasModel(defaultModel.searching()),
+            hasEffects(ParseJsonIntoPatientPrefillInfoObject(expectedJson))
+        ))
+  }
+
+  @Test
+  fun `when json is parsed then search patient with NHID`() {
+    val patientPrefillInfo: PatientPrefillInfo? = adapter.fromJson(expectedJson)
+    val indiaNationalHealthID = patientPrefillInfo!!.healthIdNumber
+
+    val identifier = Identifier(indiaNationalHealthID, Identifier.IdentifierType.IndiaNationalHealthId)
+
+    spec
+        .given(defaultModel)
+        .whenEvent(EnteredCodeJsonParsed(patientPrefillInfo))
+        .then(assertThatNext(
+            hasModel(defaultModel.searching()),
+            hasEffects(SearchPatientByIdentifier(identifier))
+        ))
+  }
+
+  @Test
+  fun `when searching for patient, then ignore newly scanned india national health Id identifier`() {
```

Comment:
I don't understand this test, can you explain what we are trying to do here, please?

Reply:
The scenario where we've scanned an ID and are searching for patient, while searching if we scan another then we ignore that ID.

Reply:
Hmm, that scenario won't happen, right? For example, when an ID is scanned and we are searching we show a progress bar and then navigate to a different screen based on the results. WDYT?

Reply:
Right. Checked ✅